src/schema/comments.ts

```diff
@@ -523,14 +561,28 @@ export const resolvers: IResolvers<any, Context> = {
     },
     recommendedMentions: async (
       _,
-      { postId, query, limit = 5 }: GQLMentionUserArgs,
+      { postId, query, limit = 5, sourceId }: GQLMentionUserArgs,
       ctx,
       info,
     ): Promise<User[]> => {
       const { con, userId } = ctx;
-      const ids = await (query
-        ? recommendUsersByQuery(con, userId, { query, limit })
-        : recommendUsersToMention(con, postId, userId, { limit }));
+      const getIds = async () => {
+        const userIds = await (query
+          ? recommendUsersByQuery(con, userId, { query, limit })
+          : recommendUsersToMention(con, postId, userId, { limit }));
+
+        if (!sourceId) {
+          return userIds;
+        }
+
+        const members: SourceMember[] = await con
+          .getRepository(SourceMember)
+          .findBy({ userId: In(userIds), sourceId });
```

Comment:
Is the `In` needed?
Does it not limit the mentions, by only recommended people in the first place?

Maybe it doesn't though.

Reply:
This is already the recommended people. We are just filtering it.